File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    const map = {}
    for (const num of nums) {
        map[num] = map[num] + 1 || 1
    }
    let res = 0
    for (const num in map) {
        if (map[num] > 1) {
            res += Math.floor((map[num] * (map[num]-1)) / 2)
        }
    }
    return res
};

numIdenticalPairs = function(nums) {
    let res = 0
    for (let i=0; i < nums.length; i++) {
        for (let j=i+1; j < nums.length; j++) {
            if (nums[i] === nums[j]) {
                res++
            }
        }
    }
    return res
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.kt
class Solution {
    fun numIdenticalPairs(nums: IntArray): Int {
        var res = 0
        for (i in 0..(nums.size-1)) {
            for (j in (i+1)..(nums.size-1)) {
                if (nums[i] == nums[j]) {
                    res += 1
                }
            }
        }
        return res
    }
}